### PR TITLE
Add person student teacher

### PR DESCRIPTION
**Pull request by Ochiengsteven**
👩‍🏫 **Feature Implementation: Person, Student, and Teacher Classes**

This pull request introduces the implementation of the `Person`, `Student`, and `Teacher` classes, each designed with specific attributes and methods as specified in the project requirements.

**Class Implementations:**

1. `Person` class:
   - Instance variables: `@id`, `@name`, and `@age`.
   - Constructor includes parameters: `name`, `age`, and `parent_permission` (optional; default values: "Unknown" and `true`).
   - Getters implemented for `@id`, `@name`, and `@age`.
   - Setters for `@name` and `@age`.
   - `of_age?` method implemented (private), returning `true` if the age is 18 or older, otherwise `false`.
   - `can_use_services?` method (public) returns `true` if the person is of age or has parental permission.

2. `Student` class (extends `Person`):
   - Inherits from `Person`.
   - Constructor extends the parent's constructor by adding `@classroom` parameter.
   - Method `play_hooky` implemented, returning "¯\(ツ)/¯".

3. `Teacher` class (extends `Person`):
   - Inherits from `Person`.
   - Constructor extends the parent's constructor by adding `@specialization` parameter.
   - `can_use_services?` method overridden to always return `true`.

**Code Organization:**

- Each class is saved in a separate file to ensure proper code organization and maintainability.

**Class Functionalities and Inheritance:**

- Each class has been structured and its functionalities are designed in accordance with the specified requirements.
- Inheritance is effectively utilized for the `Student` and `Teacher` classes, inheriting attributes and methods from the base `Person` class.

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @Ochiengsteven,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* You have submitted a project that has no descriptive readme file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, name = 'Unknown')
+    super(name)
+    self.name = name
+    @specialization = specialization
+  end
```
Comment (OyePriscilla) [inline teacher.rb]: - [ ] For class Teacher, while making a call to the `parent class`, kindly ensure that all the necessary `parameters `are initialized alongside with **`name`**.

## Project Requirement
Untitled

```suggestion
  def initialize(specialization, name = 'Unknown')
    super(age, name, parent_permission)
    self.name = name
    @specialization = specialization
  end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def can_use_service?
+    of_age || parent_permission
+  end
+
+  private
+
+  def of_age
+    age >= 18
+  end
```
Comment (OyePriscilla) [inline person.rb]: - [ ] In Ruby, a question mark (?) at the end of a method name is a naming convention that's often used to indicate that the method returns a boolean value. These methods are commonly referred to as "predicate methods." Predicate methods typically return true or false based on some condition. Kindly add **`?`** to the method name. ℹ️ 

`def of_age?`

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @id = SecureRandom.uuid
+    @parent_permission = parent_permission
+  end
+
+  def can_use_service?
+    of_age || parent_permission
+  end
```
Comment (OyePriscilla) [inline person.rb]: - [ ] A variable name with a question mark (?) at the end is often used to indicate that the variable stores a boolean value or represents a condition. Kindly add **`?`** to the variable name in line 15. ℹ️ 

`of_age?`

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Ochiengsteven 👏 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implement classes and objects in Ruby. ✔️
Implement encapsulation and inheritance with Ruby. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @Ochiengsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features as requested by the previous code reviewer ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**